hooks/pre_prompt.py

```diff
@@ -0,0 +1,63 @@
+from typing import Any, Dict, Optional
+import json
+from pathlib import Path
+from rich.prompt import Prompt
+from rich.console import Console
+
+
+def main():
+    cookiecutter_config_path = Path("cookiecutter.json")
+    if not cookiecutter_config_path.exists():
+        return
+
+    cookiecutter_json = json.loads(cookiecutter_config_path.read_text())
+    try:
+        extended_cookiecutter_json = extend_default_cookiecutter(cookiecutter_json)
+    except KeyboardInterrupt:
+        return
+
+    if extended_cookiecutter_json:
+        cookiecutter_config_path.write_text(
+            json.dumps(extended_cookiecutter_json, indent=4)
+        )
+
+
+def extend_default_cookiecutter(
+    cookiecutter_json: Dict[str, Any]
+) -> Optional[Dict[str, Any]]:
+    """
+    Extend the default cookiecutter.json with additional plugin configuration.
+
+    :param cookiecutter_json: The default cookiecutter.json
+    :return: The extended cookiecutter.json with plugin configuration
+    """
+    project_type = prompt_for_generic_or_plugin()
+
+    if project_type == "2":
+        cookiecutter_json["__project_type"] = "agent_plugin"
+        cookiecutter_plugin_config_path = Path("cookiecutter_plugin.json")
+        if cookiecutter_plugin_config_path.exists():
+            plugin_config = json.loads(cookiecutter_plugin_config_path.read_text())
+
+            return {**plugin_config, **cookiecutter_json}
```

Comment:
Should the order of these be swapped so that, if necessary, the agent plugin config can override the base cookiecutter?

Reply:
I don't think so. I expect the plugin config to come right after the default one after I choose a project type.